### PR TITLE
Add tests to validate enclave creation using conf files; fixed oesign bug when creating 0-base enclaves

### PR DESCRIPTION
**Pull request by RRathna**
Signed-off-by: Rathna Ramesh <rathna1993@gmail.com>

### CODE REVIEW

**Review by radhikaj — APPROVED** [review]

File: tests/sgx_zerobase/enc/create_zerobase.conf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+
+# Enclave settings:
+Debug=1
```
Comment (mingweishih) [inline tests/sgx_zerobase/enc/create_zerobase.conf:5]: Shouldn't the conf file specifies the 0-base options?

**Review by mingweishih — COMMENTED** [review]
Can you make sure the following cases are cover
- conf enables the 0-base feature regardless whether the macro sets or not.
- conf disable the 0-base feature (or simply does not specify the 0-base option) regardless whether the macro sets or not.

File: tests/sgx_zerobase/enc/create_zerobase.conf (hunk 1 above)
Comment (RRathna) [reply tests/sgx_zerobase/enc/create_zerobase.conf]: Modified the test cases to address these scenarios.

File: tests/sgx_zerobase/CMakeLists.txt [hunk 2]
```diff
@@ -7,4 +7,21 @@ if (BUILD_ENCLAVES)
...
+
+# Tests 0-base enclave creation when OE_SET_ENCLAVE_SGX*() macro disables
+# 0-base enclave creation and oesign configuration file enables 0-base enclave creation.
+add_enclave_test(
+  tests/sgx_zerobase_prop_macro_disable_conf_enable sgx_zerobase_host
+  sgx_zerobase_macro_disable_conf_enable_enc_signed)
+
+# Tests 0-base enclave creation when OE_SET_ENCLAVE_SGX*() macro enables
+# 0-base enclave creation but oesign configuration file disables 0-base enclave creation.
+add_enclave_test(tests/sgx_zerobase_macro_enable_conf_disable sgx_zerobase_host
```
Comment (mingweishih) [inline tests/sgx_zerobase/CMakeLists.txt:26]: will the test be bypassed in this case given that the zero-base is not enabled? I didn't see relevant check on the host.

**Review by mingweishih — APPROVED** [review]
LGTM. Just have one question on the case of `sgx_zerobase_macro_enable_conf_disable`.

File: tests/sgx_zerobase/CMakeLists.txt (hunk 2 above)
Comment (RRathna) [reply tests/sgx_zerobase/CMakeLists.txt:26]: > will the test be bypassed in this case given that the zero-base is not enabled? I didn't see relevant check on the host.

Just added a check in host/host.cpp to return 0 when it sees a "config_disable" and not run the 0-base specific tests.

**Review by anakrish — APPROVED** [review]

**Pull request merged**